### DISCUSSION TITLE
Different quality setting for different areas of an image

### DISCUSSION THREAD

**Discussion by tefod-zz** [ImageMagick/ImageMagick#2393] [Help]
Hi community,

regarding [this](http://www.imagemagick.org/discourse-server/viewtopic.php?t=24234) thread in the forum from 2013 I am trying the whole to apply a lower compression to a specific area of an image.. I gave up :|

I have a picture 800x600, where one half should compressed much more than the other

`convert.exe -respect-parentheses ./#pics_all/Pic.jpg 
( -clone 0 -quality 5% +write PicLow.jpg ) +delete PicLow.jpg
( -clone 0 -fill white -colorize 100 -fill black -draw "rectangle 0,0 800,300" )
-compose over -composite ttt2.webp`

It doesnt work allthough a tried all possible variations (with quality settings on the end, with jpg etc.)

Any advices?
Thanks.

**Comment by snibgo** [comment]
What version of IM, on what platform? Do you get any error messages?

Your command is bad for both bash and Windows BAT.

**Comment by tefod-zz** [comment]
Hey,
thank you so much; yeap, sure.. Still learning imagemagick; amazing how powerful it is.
I am using v7.0.10.19 Q8 x86 with Win 10 x64.

I try to explain, what my goal is. I developed a [powershell tool](https://abload.de/gallery.php?key=ktC2ybr0), which is recursively scanning/converting a directory, manipulation the pictures and reverse injects them into a [datatables ](https://datatables.net/) HTML file.
For beta releases, i will watermark the (>1000 pictures), add some label and a barcode (looks cool).
Example:
[This](https://abload.de/img/sourceaaknb.jpg) is the source. [This](https://abload.de/img/targetm7jof.jpg) is the target.

Done with following command (ignore the line breaks):
`imagemagick.exe ./source.jpg  
-gravity north -chop 0x1 -gravity east -chop 1x0 -gravity south -chop 0x1 -gravity west -chop 1x0  
-resize 800> -strip -quality 75% 
( -tile-offset +50+35 -background none -fill rgba(197,197,197,0.25) -pointsize 24 label:"Draft 20200807" -rotate -30 -write mpr:my_text +delete )  
( +clone -tile mpr:my_text -draw "color 0,0 reset" )  
-compose xor  
-composite -compose over -border 4  
-background white -gravity south -splice 0x52+0+0  
-font code128.ttf -pointsize 40 -gravity southeast -fill black -annotate 0x0-2+10 "Ñhttps://github.com~Ó"  
-gravity south -chop 0x30  
-font Lucida-Console -pointsize 12 -gravity southwest -fill rgba(0,0,0,0.4) -annotate 0x15+4-0 "Some Text" 
-background white -gravity south -splice 0x6+0+0 
target.jpg`

This is running pretty flawlessly good - especially with webp. 
Question now: is there a meaningful way, to compress the upper area with e.g. 35% and the bottom-footer e.g. with 80%?

Regards...

**Comment by snibgo** [comment]
> Question now: is there a meaningful way, to compress the upper area with e.g. 35% and the bottom-footer e.g. with 80%?

You can certainly apply a visual effect to just part of the image. That effect could be a blur, or a JPEG distortion, or anything you want.

Then you can save the result to a lossless PNG, or lossy JPEG, or whatever you want.

But when you save a JPEG file, the "-quality" setting will apply to the entire image.
